AspNetCoreApplication/Services/AuthenticationService.cs

```diff
@@ -23,10 +25,18 @@ public class AuthenticationService : IAuthenticationService
 
         private readonly DataContext dataContext;
 
-        public AuthenticationService(TokenConfig tokenConfig, DataContext dataContext)
+        private readonly IHttpContextAccessor httpContextAccessor;
+
+        public AuthenticationService(TokenConfig tokenConfig, DataContext dataContext, IHttpContextAccessor httpContextAccessor)
         {
             this.tokenConfig = tokenConfig;
             this.dataContext = dataContext;
+            this.httpContextAccessor = httpContextAccessor;
+        }
+
+        public UserClaimsPrincipal GetCurrentUser()
+        {
+            return (UserClaimsPrincipal)httpContextAccessor.HttpContext.User;
```

Comment:
```suggestion
        ```suggestion
        public AuthenUser GetCurrentUser()
        {
            return ((UserClaimsPrincipal)httpContextAccessor.HttpContext.User).AuthenUser;
```

Reply:
done